core_backend/app/tags/models.py

```diff
@@ -0,0 +1,158 @@
+from datetime import datetime
+from typing import List, Optional
+
+from sqlalchemy import (
+    Column,
+    DateTime,
+    ForeignKey,
+    Integer,
+    String,
+    Table,
+    delete,
+    select,
+)
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from ..models import Base
+from .schemas import TagCreate
+
+content_tags_table = Table(
+    "content_tags",
+    Base.metadata,
+    Column("content_id", Integer, ForeignKey("content.content_id"), primary_key=True),
+    Column("tag_id", Integer, ForeignKey("tag.tag_id"), primary_key=True),
+)
+
+
+class TagDB(Base):
+    __tablename__ = "tag"
+
+    tag_id: Mapped[int] = mapped_column(Integer, primary_key=True, nullable=False)
+    tag_name: Mapped[str] = mapped_column(String(length=50), nullable=False)
+    created_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
+    updated_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
+    contents = relationship(
+        "ContentDB", secondary=content_tags_table, back_populates="content_tags"
+    )
+    user_id: Mapped[int] = mapped_column(
```

Comment:
can we move this to be just after "tag_id"? It's an important id col so lets keep it at the start of the table. Might have to move the migration line too